app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,471 @@
+    });
+
+    builder.setTitle(getString(R.string.hotspot_dialog_title));
+    builder.setMessage(
+        getString(R.string.hotspot_dialog_message)
+    );
+    AlertDialog dialog = builder.create();
+    dialog.show();
+  }
+
+  //Keeps checking if hotspot has been turned using the ip address with an interval of 1 sec
+  //If no ip is found after 15 seconds, dismisses the progress dialog
+  private void pollForValidIpAddress() {
+    Flowable.fromCallable(() -> ServerUtils.getIp())
+        .retryWhen(error -> error.delay(1, TimeUnit.SECONDS))
+        .timeout(15, TimeUnit.SECONDS)
+        .firstOrError()
+        .observeOn(AndroidSchedulers.mainThread())
+        .subscribe(new SingleObserver<String>() {
+          @Override public void onSubscribe(Disposable d) {
+          }
+
+          @Override public void onSuccess(String s) {
+            progressDialog.dismiss();
+            startService(createHotspotIntent(ACTION_START_SERVER).putStringArrayListExtra(
+                SELECTED_ZIM_PATHS_KEY, selectedBooksPath));
+            Log.d(TAG, "onSuccess:  " + s);
+          }
+
+          @Override public void onError(Throwable e) {
+            // display the ip and don't forget to dismiss the dialog
+            progressDialog.dismiss();
+            Toast.makeText(ZimHostActivity.this, R.string.server_failed_message, Toast.LENGTH_SHORT)
+                .show();
+            Log.d(TAG, "Unable to turn on server", e);
+          }
+        });
+  }
+
+  Intent createHotspotIntent(String action) {
+    return new Intent(this, HotspotService.class).setAction(action);
+  }
+
+  void mobileDataDialog() {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+      AlertDialog.Builder builder = new AlertDialog.Builder(this, dialogStyle());
+
+      builder.setPositiveButton(this.getString(R.string.yes),
+          (dialog, id) -> openMobileDataActivity());
+      builder.setNegativeButton((android.R.string.no),
+          (dialog, id) -> startHotspotManuallyDialog());
+      builder.setTitle(this.getString(R.string.mobile_data_enabled));
+      builder.setMessage(
+          this.getString(R.string.mobile_data_message) + "\n" + this.getString(
+              R.string.mobile_data_message_confirmation)
+      );
+      builder.setCancelable(false);
+      AlertDialog dialog = builder.create();
+      dialog.show();
+    }
+  }
+
+  @Override public void onServerStarted(@NonNull String ipAddress) {
+    this.ip = ipAddress;
+    layoutServerStarted();
+  }
+
+  @Override public void onServerStopped() {
+    layoutServerStopped();
+    if (selectedBooksPath.size() > 0) {
+      selectedBooksPath.clear();
+    }
+  }
+
+  @Override public void onServerFailedToStart() {
+    Toast.makeText(this, R.string.server_failed_toast_message, Toast.LENGTH_LONG).show();
+  }
+
+  @Override public void onHotspotTurnedOn(@NonNull WifiConfiguration wifiConfiguration) {
+
+    hotspotService.startForegroundNotificationHelper();
```

Comment:
you shouldn't have to tell the hotspotService to do this, it already received this event so it could do this itself

Reply:
Fixed. 6f13e64fee3aa9b0dc9448243b2a7072d60ed78b